app/components/root/root.js

```diff
@@ -29,21 +40,98 @@ export default class Root extends React.Component {
     componentDidMount() {
         AppState.addEventListener('change', this.handleAppStateChange);
         EventEmitter.on(Constants.CONFIG_CHANGED, this.handleConfigChanged);
+        MessageBarManager.registerMessageBar(this.refs.notification);
+        this.configurePushNotifications();
     }
 
     componentWillUnmount() {
         AppState.removeEventListener('change', this.handleAppStateChange);
         EventEmitter.off(Constants.CONFIG_CHANGED, this.handleConfigChanged);
+        MessageBarManager.unregisterMessageBar();
+        PushNotification.unregister();
     }
 
+    onRegisterDevice = (data) => {
```

Comment:
I wonder if it would be a good idea to move the code for handling push notifications into some `PushNotifications` component that gets mounted as part of the `Root` component, but always renders null? That would help keep the code for that nicely encapsulated while still having it available in every scene. It could also wrap the `MessageBar` and return that so even that is together with the rest of the code